tests/profiling/collector/test_threading.py

```diff
@@ -312,3 +313,40 @@ def test_lock_acquire_release_speed_patched(benchmark, pct):
 )
 def test_lock_acquire_release_speed(benchmark):
     benchmark(_lock_acquire_release, threading.Lock())
+
+
+@pytest.mark.skipif(not sys.platform.startswith("linux"), reason="only works on linux")
+@pytest.mark.subprocess(
+    ddtrace_run=True,
+    env=dict(DD_PROFILING_ENABLED="true"),
+    err=None,
+)
+def test_user_threads_have_native_id():
+    from os import getpid
+    from threading import Thread
+    from threading import _MainThread
+    from threading import current_thread
+    from time import sleep
+
+    main = current_thread()
+    assert isinstance(main, _MainThread)
+    # We expect the main thread to have the same ID as the PID
+    assert main.native_id == getpid(), (main.native_id, getpid())
```

Comment:
+1 to marking this test Linux-only, as this assumption in particular obviously breaks on other platforms (even macos).